### PR TITLE
feat(sonarr): add notification

### PR DESCRIPTION
**Pull request by Fuochi**
- add notification

### CODE REVIEW

File: sonarr/notification.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+	OnEpisodeFileDeleteForUpgrade bool                `json:"onEpisodeFileDeleteForUpgrade"`
+	OnHealthIssue                 bool                `json:"onHealthIssue"`
+	OnApplicationUpdate           bool                `json:"onApplicationUpdate"`
+	IncludeHealthWarnings         bool                `json:"includeHealthWarnings"`
+	ID                            int64               `json:"id,omitempty"`
+	Name                          string              `json:"name"`
+	Implementation                string              `json:"implementation"`
+	ConfigContract                string              `json:"configContract"`
+	Tags                          []int               `json:"tags"`
+	Fields                        []*starr.FieldInput `json:"fields"`
```
Comment (davidnewhall) [inline sonarr/notification.go:33]: I'm sure glad this works.

File: sonarr/notification.go [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+	return s.GetNotificationsContext(context.Background())
+}
+
+// GetNotificationsContext returns all configured notifications.
+func (s *Sonarr) GetNotificationsContext(ctx context.Context) ([]*NotificationOutput, error) {
+	var output []*NotificationOutput
+
+	req := starr.Request{URI: bpNotification}
+	if err := s.GetInto(ctx, req, &output); err != nil {
+		return nil, fmt.Errorf("api.Get%s): %w", &req, err)
```
Comment (davidnewhall) [inline sonarr/notification.go:78]: missing paren

File: sonarr/notification.go [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+	return s.GetNotificationContext(context.Background(), notificationID)
+}
+
+// GetNotificationContext returns a single notification.
+func (s *Sonarr) GetNotificationContext(ctx context.Context, notificationID int) (*NotificationOutput, error) {
+	var output NotificationOutput
+
+	req := starr.Request{URI: path.Join(bpNotification, fmt.Sprint(notificationID))}
+	if err := s.GetInto(ctx, req, &output); err != nil {
+		return nil, fmt.Errorf("api.Get%s): %w", &req, err)
```
Comment (davidnewhall) [inline sonarr/notification.go:95]: another... looks like you missed a bunch.

File: sonarr/notification.go [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+	client *NotificationInput,
+) (*NotificationOutput, error) {
+	var output NotificationOutput
+
+	var body bytes.Buffer
+	if err := json.NewEncoder(&body).Encode(client); err != nil {
+		return nil, fmt.Errorf("json.Marshal(%s): %w", bpNotification, err)
+	}
+
+	req := starr.Request{URI: path.Join(bpNotification, fmt.Sprint(int(client.ID))), Body: &body}
```
Comment (davidnewhall) [inline sonarr/notification.go:139]: `int()` is not needed.

**Review by davidnewhall — CHANGES_REQUESTED** [review]

**Comment by davidnewhall** [comment]
Godsend if you add this to the others and close #62. 💚

File: sonarr/notification.go (hunk 3 above)
Comment (Fuochi) [reply sonarr/notification.go:95]: yep, it seems I did few find and replace with wrong replacement

**Comment by Fuochi** [comment]
> Godsend if you add this to the others and close #62. 💚

I've few more branches locally that I have to push before starting working on something else, maybe around end of october?

File: sonarr/notification_test.go [hunk 5]
```diff
@@ -0,0 +1,554 @@
...
+	`"onSeriesDelete":false,"onEpisodeFileDelete":true,"onEpisodeFileDeleteForUpgrade":false,` +
+	`"onHealthIssue":false,"onApplicationUpdate":false,"includeHealthWarnings":false,"name":"Test",` +
+	`"implementation":"CustomScript","configContract":"CustomScriptSettings","tags":null,` +
+	`"fields":[{"name":"path","value":"/scripts/sonarr.sh"},{"name":"apiPath","value":"/api"}]}`
+
+const updateNotification = `{"onGrab":false,"onDownload":true,"onUpgrade":true,"onRename":false,` +
+	`"onSeriesDelete":false,"onEpisodeFileDelete":true,"onEpisodeFileDeleteForUpgrade":false,` +
+	`"onHealthIssue":false,"onApplicationUpdate":false,"includeHealthWarnings":false,"id":3,"name":"Test",` +
+	`"implementation":"CustomScript","configContract":"CustomScriptSettings","tags":null,` +
+	`"fields":[{"name":"path","value":"/scripts/sonarr.sh"},{"name":"apiPath","value":"/api"}]}`
```
Comment (davidnewhall) [inline sonarr/notification_test.go:74]: This is a lot of damn work.

**Comment by Fuochi** [comment]
so... I changed back the `omitempty` tag to be used with booleans as well, during my last review I noticed that I was mistaken when I thought that those fields aren't working. we just have to remember that missing=false, even for `PUT`.

This should work. Unfortunately I won't have much time in the next weeks to help you with this project. If you're ok, we can start by merging this, then we'll see for the other *arrs

**Review by davidnewhall — APPROVED** [review]
I'm good with this. We can iron out problems later. I'll be using this code soon enough to update naming profiles in the wild.

**Pull request merged**